modules/gapi/src/compiler/passes/transformations.cpp

```diff
@@ -21,33 +21,55 @@
 namespace cv { namespace gimpl { namespace passes {
 namespace
 {
-// Generates ADE graph from expression-based computation
-std::unique_ptr<ade::Graph> makeGraph(const cv::GComputation& c)
+using Graph = GModel::Graph;
+using Metadata = typename Graph::CMetadataT;
+
+// Checks pairs of {pattern node, substitute node} and asserts if there are any incompatibilities
+void checkDataNodes(const Graph& pattern,
+                    const Graph& substitute,
+                    const std::vector<ade::NodeHandle>& patternNodes,
+                    std::vector<ade::NodeHandle> substituteNodes)
```

Comment:
the last parameter should be `const &` as well?

Reply:
indeed. thanks!